admin-dev/themes/new-theme/js/components/module-card.ts

```diff
@@ -428,7 +428,10 @@ export default class ModuleCard {
           mainElement.attr('data-installed', '0');
           mainElement.attr('data-active', '0');
 
+          refreshNeeded = true; // if uninstalled, some sub pages may change (updates)
           BOEvent.emitEvent('Module Uninstalled', 'CustomEvent', mainElement);
+        } else if (action === 'update') {
+          refreshNeeded = true; // if module upgraded, sub pages may change (updates)
```

Comment:
I think this should be handled server side: https://github.com/PrestaShop/PrestaShop/blob/8.0.x/src/PrestaShopBundle/Controller/Admin/Improve/ModuleController.php#L214

Reply:
Do you mean that I should add to ModuleController something like :

```php
if ($action === 'uninstall') {
    $response[$module]['refresh_needed'] = true;
}
```
and replace 
```php
$response[$module]['refresh_needed'] = $this->moduleNeedsReload($moduleInstance);
```
by
```php
$response[$module]['refresh_needed'] = true;
```
under `update` action ?

I don't understand `moduleNeedsReload` use case in fact ? How can I know if the frontend that requested the action needs a reload or not without considering the context ? I mean that in our case we have a list that should be updated. What is the link with the module having hooks or not ? Should I change `moduleNeedsReload` to integrate some front-end context and build it's answer depending on it ?

Reply:
Discussed with @atomiix , I'll remove this code. If a list refresh is required for update/uninstall actions, it should be a dedicated issue and it may induce some architecture analysis / discussions.